File: LiveDataSampleJava/app/src/main/java/com/websarva/wings/android/livedatasamplejava/data/CocktailmemoRepository.java
package com.websarva.wings.android.livedatasamplejava.data;

import android.app.Application;

import com.google.common.util.concurrent.ListenableFuture;
import com.websarva.wings.android.livedatasamplejava.data.local.AppDatabase;
import com.websarva.wings.android.livedatasamplejava.data.local.Cocktailmemo;
import com.websarva.wings.android.livedatasamplejava.data.local.CocktailmemoDAO;

import java.util.concurrent.ExecutionException;

public class CocktailmemoRepository {
	private AppDatabase _db;
	public CocktailmemoRepository(Application application) {
		_db = AppDatabase.getDatabase(application);
	}

	public Cocktailmemo getCocktailmemo(int cocktailId) {
		CocktailmemoDAO cocktailmemoDAO = _db.createCocktailmemoDAO();
		ListenableFuture<Cocktailmemo> future = cocktailmemoDAO.findByPK(cocktailId);
		Cocktailmemo cocktailmemo = null;
		try {
			cocktailmemo = future.get();
		}
		catch(ExecutionException ex) {
			ex.printStackTrace();
		}
		catch(InterruptedException ex) {
			ex.printStackTrace();
		}
		return cocktailmemo;
	}

	public boolean deleteInsertCocktailmemo(Cocktailmemo cocktailmemo) {
		CocktailmemoDAO cocktailmemoDAO = _db.createCocktailmemoDAO();
		ListenableFuture<Integer> futureDelete = cocktailmemoDAO.delete(cocktailmemo);
		boolean result = false;
		try {
			int resultDelete = futureDelete.get();
			if(resultDelete >= 0) {
				ListenableFuture<Long> futureInsert = cocktailmemoDAO.insert(cocktailmemo);
				long resultInsert = futureInsert.get();
				if(resultInsert >= 0) {
					result = true;
				}
			}
		}
		catch(ExecutionException ex) {
			ex.printStackTrace();
		}
		catch(InterruptedException ex) {
			ex.printStackTrace();
		}
		return result;
	}
}

File: LiveDataSampleKotlin/app/src/main/java/com/websarva/wings/android/livedatasamplekotlin/data/CocktailmemoRepository.kt
package com.websarva.wings.android.livedatasamplekotlin.data

import android.app.Application
import com.websarva.wings.android.repositorysamplekotlin.data.local.AppDatabase
import com.websarva.wings.android.repositorysamplekotlin.data.local.Cocktailmemo

class CocktailmemoRepository(application: Application) {
	private val _db: AppDatabase

	init {
		_db = AppDatabase.getDatabase(application)
	}

	suspend fun getCocktailmemo(cocktailId: Int): Cocktailmemo? {
		val cocktailmemoDAO = _db.createCocktailmemoDAO()
		return cocktailmemoDAO.findByPK(cocktailId)
	}

	suspend fun deleteInsertCocktailmemo(cocktailmemo: Cocktailmemo): Boolean {
		val cocktailmemoDAO = _db.createCocktailmemoDAO()
		var result = false
		val resultDelete = cocktailmemoDAO.delete(cocktailmemo)
		if(resultDelete >= 0) {
			val resultInsert = cocktailmemoDAO.insert(cocktailmemo)
			if(resultInsert >= 0) {
				result = true
			}
		}
		return result
	}
}
